internal/information/build_info.go

```diff
@@ -0,0 +1,72 @@
+package information
+
+import (
+	"runtime/debug"
+
+	"github.com/golang/glog"
+)
+
+type BuildInfo struct {
+	commitSha       string
+	architecture    string
+	vcsTime         string
+	vcsType         string
+	operatingSystem string
+	goVersion       string
+}
+
+// GetBuildInfo returns the current Build Information as a BuildInfo type (struct) using a built-in debug package and its ReadBuildInfo() function.
+// Upon error, an empty struct is returned along with a logged error.
+func GetBuildInfo() *BuildInfo {
+	b := new(BuildInfo)
+	info, ok := debug.ReadBuildInfo()
+	if !ok {
+		glog.Fatal("unable to get build info")
+	}
+	b.goVersion = info.GoVersion
+	for _, i := range info.Settings {
+		switch i.Key {
+		case "vcs.time":
+			b.vcsTime = i.Value
+		case "GOARCH":
+			b.architecture = i.Value
+		case "vcs":
+			b.vcsType = i.Value
+		case "GOOS":
+			b.operatingSystem = i.Value
+		case "vcs.revision":
+			b.commitSha = i.Value
+		}
+	}
+	return b
+}
+
+// GetCommitSha returns the current commitSha (version).
+func (b *BuildInfo) GetCommitSha() string {
+	return b.commitSha
+}
+
+// GetArchitecture returns the current architecture.
+func (b *BuildInfo) GetArchitecture() string {
+	return b.architecture
+}
+
+// GetVCSTime returns the current commit time.
+func (b *BuildInfo) GetVCSTime() string {
```

Comment:
We should document that it is possible that this might be empty depending on whether it's been built using a vcs